app/models/order.rb

```diff
@@ -0,0 +1,18 @@
+class Order < ApplicationRecord
+  has_many :orders_line_items
+  has_many :line_items, through: :orders_line_items
+
+  before_create :add_total
+
+  scope :within_range, -> (from, to) { where(created_at: from..to) }
+
+  def total
+    read_attribute(:total).to_f if read_attribute(:total)
+  end
+
+  private
+
+  def add_total
+    self.total = line_items.map(&:total).sum.to_f
+  end
```

Comment:
Another benefit of leverage our line item class is calculating our order total. We'd have to either essentially duplicate this structure somewhere else (be it the order table or otherwise) or find a different way of calculating our total.

The use of `map` over something more efficient such as `pluck` or the `sum` database helper (which would sum all of the applicable `total` column values within the line items table) is due to the `line_items` reference not being available yet in the database (instead only on the ruby line item ruby instance.